src/main/java/org/openrewrite/staticanalysis/RemoveToStringCallsFromArrayInstances.java

```diff
@@ -75,9 +78,17 @@ public J visitMethodInvocation(J.MethodInvocation mi, ExecutionContext ctx) {
                 return buildReplacement(select, mi);
             } else if (METHOD_MATCHERS.stream().anyMatch(matcher -> matcher.matches(mi))) {
                 // deals with edge cases where .toString() is called implicitly
+                JavaType.Method methodType = mi.getMethodType();
+                if (methodType == null) {
+                    return mi;
+                }
+                List<JavaType> parameterTypes = methodType.getParameterTypes();
```

Comment:
An alternative would be to rely on `Flag.Varargs` which when set on the `JavaType.Method` means that it is a varargs method.

Reply:
Didn't know that flag, thanks! But then that only applies to the entire method right? Yet we'd still want to do replacements on String.format and friends when we can. Say String.format gets called with 6 arguments, one of them is an array. The current implementation would then still wrap that one array in `Arrays.toString`. Or how would you apply that flag here?

Reply:
Not sure. I would probably just use the flag to check what the recipe should do with the arguments corresponding to the (last) varargs parameter, if any. Probably no huge improvement...